pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -140,237 +180,133 @@ func (Plugin) OnInboundRouteConfiguration(in *plugin.InputParams, routeConfigura
+					DisableCheckCalls: true, /* TODO: enable client-side checks */
+					MixerAttributes: &mpb.Attributes{Attributes: attributes{ // TODO: fall through destination servicec
+						"destination.service": attrStringValue(UnknownDestination),
+					}},
 				},
 			},
-		},
-		ServiceConfigs: map[string]*mccpb.ServiceConfig{},
-		Transport:      transport,
-	}
-
-	var labels map[string]string
-	// Note: instances are all running on mode.Node named 'role'
-	// So instance labels are the workload / Node labels.
-	if len(nodeInstances) > 0 {
-		labels = nodeInstances[0].Labels
-		mxConfig.DefaultDestinationService = nodeInstances[0].Service.Hostname.String()
-	}
-
-	if !outboundRoute || role.Type == model.Router {
-		// for outboundRoutes there are no default MixerAttributes except for gateway.
-		// specific MixerAttributes are in per route configuration.
-		v1.AddStandardNodeAttributes(mxConfig.MixerAttributes.Attributes, v1.AttrDestinationPrefix, role.IPAddress, role.ID, labels)
-		mxConfig.MixerAttributes.Attributes["context.reporter.local"] = &mpb.Attributes_AttributeValue{
-			Value: &mpb.Attributes_AttributeValue_BoolValue{BoolValue: true},
-		}
+			Transport: buildTransport(mesh, attrUID(node)),
+		}),
 	}
+}
 
-	if role.Type == model.Sidecar && !outboundRoute {
-		// Don't forward mixer attributes to the app from inbound sidecar routes
-	} else {
-		mxConfig.ForwardAttributes = &mpb.Attributes{
-			Attributes: map[string]*mpb.Attributes_AttributeValue{},
-		}
-		addStandardNodeAttributes(mxConfig.ForwardAttributes.Attributes, v1.AttrSourcePrefix, role.IPAddress, role.ID, labels)
+func buildInboundHTTPFilter(mesh *meshconfig.MeshConfig, node *model.Proxy, attrs attributes, instances []*model.ServiceInstance) *http_conn.HttpFilter {
+	// TODO(rshriram): pick a single service for TCP destination workload. This needs to be removed once mixer config resolution
+	// can scope by workload namespace. The choice of destination service should not be made here!
+	destination := UnknownDestination
+	if len(instances) > 0 {
+		destination = instances[0].Service.Hostname.String()
 	}
+	serviceAttrs := make(attributes)
+	addDestinationServiceAttributes(serviceAttrs, destination, node.Domain)
 
-	// gateway case is special because upstream listeners are considered outbound, however we don't want to
-	// automatically disable policy / report.
-	disablePolicy := (outboundRoute && role.Type != model.Router) || mesh.DisablePolicyChecks
-	disableReport := outboundRoute && role.Type != model.Router
-
-	for _, instance := range nodeInstances {
-		mxConfig.ServiceConfigs[instance.Service.Hostname.String()] =
-			serviceConfig(instance.Service.Hostname.String(), instance, config, disablePolicy, disableReport, role.Domain)
+	return &http_conn.HttpFilter{
+		Name: mixer,
+		Config: util.MessageToStruct(&mccpb.HttpClientConfig{
+			DefaultDestinationService: UnknownDestination,
+			ServiceConfigs: map[string]*mccpb.ServiceConfig{
+				UnknownDestination: {
```

Comment:
arent you assigning destination above? Why assign to unknown destination here?

Reply:
i have the same question. I see that we are adding the `serviceAttrs` in the MixerAttributes section, but isn't the `ServiceConfigs` map still relevant?

Reply:
Service config has append semantics. We want to have common attributes in `attrs` and destination service-specific in service configs. Then the result is the combination. Mixer/mixerfilter do require something to be set for destination.service.